### PR TITLE
Add IBS

### PR DESCRIPTION
**Pull request by Kiri2002**
add IBS: Illustrator for Biology

## Involved packages

 - IBS

## Involved issue

Close #

## Details

- [x] Tested in the local machine (largest 16G RAM) and it is passed without any issue
- [x] Provide New Package
- [ ] Fix the Packages
  - [ ] PKGBUILD
  - [ ] lilac.yaml
  - [ ] lilac.py
- [x] Would like to continue to work with us

### CODE REVIEW

File: BioArchLinux/ibs/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Maintainer: kiri <kiri@vern.cc>
+pkgname=ibs
+_pkgname=IBS
+pkgver=1.0.3
+pkgrel=1
+pkgdesc="Illustrator for Biology"
+arch=('any')
+url="https://ibs.renlab.org"
+license=('custom:commercial')
+depends=('java-environment=8' 'bash')
```
Comment (sukanka) [inline BioArchLinux/ibs/PKGBUILD:10]: Do you really need `java-environment`?  Have you read https://wiki.archlinux.org/title/Java_package_guidelines#Dependencies? I recommend to use `java-runtime`

File: BioArchLinux/ibs/IBS.desktop [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env xdg-open
+[Desktop Entry]
+Type=Application
+Name=IBS
+Exec=env PATH=/usr/lib/jvm/java-8-openjdk/jre/bin/:$PATH /bin/sh /opt/IBS/"IBS 1.0.3"
+Icon=/opt/IBS/.install4j/i4j_extf_2_ocjbc0_c9fozv.png
```
Comment (sukanka) [inline BioArchLinux/ibs/IBS.desktop:6]: please rename the icon and put it to someplace like `/usr/share/icons/hicolor/256x256/apps` (depends on the resolution of the icon), and just use `Icon=IBS`

File: BioArchLinux/ibs/PKGBUILD (hunk 1 above)
Comment (Kiri2002) [reply BioArchLinux/ibs/PKGBUILD:10]: i totally agree with u! i will change now

**Comment by sukanka** [comment]
At the very beginning, you should make your title meaning ful.

File: BioArchLinux/ibs/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+source=("${_pkgname}-${pkgver}.tar.gz::http://ibs.biocuckoo.org/download/IBS_1.0.3_unix_20170614.tar.gz"
+        "${_pkgname}.desktop")
+noextract=("${_pkgname}.jar")
```
Comment (sukanka) [inline BioArchLinux/ibs/PKGBUILD:13]: This line is not needed, since no `jar` file is provided in `source`

File: BioArchLinux/ibs/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+sha256sums=('2d8db720b5071f534cd49a1a29f6f5baa81966e436e5d19c492d08dfb42dcfbe'
+            '6f447dd862c3a871a64a8241eaff0b1b88114dadb8ba8aca86c67dbaf351936d')
+package() {
+	mkdir -p ${pkgdir}/opt/${_pkgname}
+	tar -p -xvf ${_pkgname}-${pkgver}.tar.gz
```
Comment (sukanka) [inline BioArchLinux/ibs/PKGBUILD:18]: Why do you extract the tarball here, it should be extracted automatically.

**Comment by sukanka** [comment]
Another thing to mention is that, we (maybe just I) prefer files to be placed under `/usr` instead of `/opt`, unless it's really a mess. 

As it is a java package, I strongly recommend you to follow [Java package guidelines](https://wiki.archlinux.org/title/Java_package_guidelines#Dependencies)

**Comment by Kiri2002** [comment]
> Another thing to mention is that, we (maybe just I) prefer files to be placed under `/usr` instead of `/opt`, unless it's really a mess.
> 
> As it is a java package, I strongly recommend you to follow [Java package guidelines](https://wiki.archlinux.org/title/Java_package_guidelines#Dependencies)
i tried, but relly mass: it has a .jar files lib, and a folder with icons but also 2 jar,and some picture folders to using to make pictures. actually it's very mass,and it start with it own sh! so i just sent it to /opt

**Comment by starsareintherose** [comment]
> Another thing to mention is that, we (maybe just I) prefer files to be placed under `/usr` instead of `/opt`, unless it's really a mess.
> 
> As it is a java package, I strongly recommend you to follow [Java package guidelines](https://wiki.archlinux.org/title/Java_package_guidelines)

Also am I put it under /usr instead of /opt, I have packaged one pkg called pleroma which has quite a lot lib and so on, then I put it under /opt/pleroma, it's a messy path.

File: BioArchLinux/ibs/IBS.desktop [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env xdg-open
+[Desktop Entry]
+Type=Application
+Name=IBS
+Exec=env PATH=/usr/lib/jvm/java-8-openjdk/jre/bin/:$PATH /bin/sh /usr/share/ibs/IBS
```
Comment (sukanka) [inline BioArchLinux/ibs/IBS.desktop:5]: So `Exec=ibs` will be enough for this

File: BioArchLinux/ibs/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+pkgver=1.0.3
+pkgrel=1
+pkgdesc="Illustrator for biological sequences"
+arch=('any')
+url="https://ibs.renlab.org"
+license=('custom:commercial')
+depends=('java-runtime=8' 'bash')
+source=("${_pkgname}-${pkgver}.tar.gz::http://ibs.biocuckoo.org/download/IBS_1.0.3_unix_20170614.tar.gz"
+        "${_pkgname}.desktop"
+        "$pkgname")
```
Comment (sukanka) [inline BioArchLinux/ibs/PKGBUILD:13]: Consider add `ibs` to source array

File: BioArchLinux/ibs/ibs [hunk 7]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/sh
+export  PATH=/usr/lib/jvm/java-8-openjdk/jre/bin/:$PATH
+exec  /usr/share/ibs/IBS "$@"
```
Comment (sukanka) [inline BioArchLinux/ibs/ibs:3]: I recommend to rename it to `ibs.sh`.

File: BioArchLinux/ibs/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+sha256sums=('2d8db720b5071f534cd49a1a29f6f5baa81966e436e5d19c492d08dfb42dcfbe'
+            '40f93f23012d471073948dd2b7ca26d444f407b02a93f92c57f734b5fda5a3fd'
+            '2f63e064f153d80f5700088aede92707c0c02353e166e991747f2b4fbd9f8927')
+package() {
+	mkdir -p $pkgdir/usr/share/$pkgname
```
Comment (sukanka) [inline BioArchLinux/ibs/PKGBUILD:18]: and in `package()` add `install -Dm755 ${srcdir}/ibs.sh  ${pkgdir}/usr/bin/ibs`

File: BioArchLinux/ibs/IBS.desktop [hunk 9]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env xdg-open
```
Comment (sukanka) [inline BioArchLinux/ibs/IBS.desktop:1]: Shebang is not needed in desktop files according to my knowledge. If I am wrong , please show me a reference

File: BioArchLinux/ibs/PKGBUILD [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+pkgname=ibs
+_pkgname=IBS
+pkgver=1.0.3
+pkgrel=1
+pkgdesc="Illustrator for biological sequences"
+arch=('any')
+url="https://ibs.renlab.org"
+license=('custom:commercial')
+depends=('java-runtime=8' 'bash')
+source=("${_pkgname}-${pkgver}.tar.gz::http://ibs.biocuckoo.org/download/IBS_1.0.3_unix_20170614.tar.gz"
```
Comment (sukanka) [inline BioArchLinux/ibs/PKGBUILD:11]: the download url is static, I recommend you to replace it with 
```
"${_pkgname}-${pkgver}.tar.gz::http://ibs.biocuckoo.org/download/IBS_${pkgver}_unix_20170614.tar.gz"
```
or even with 
```
"${_pkgname}-${pkgver}.tar.gz::http://ibs.biocuckoo.org/download/IBS_${pkgver}.tar.gz"
```
(and then `pkgver` becomes `1.0.3_unix_20170614`

**Comment by Kiri2002** [comment]
OK, changed to /user, but not isolating the '.jar' files, because the '.sh' start need to be all dependens in one road.

File: BioArchLinux/ibs/lilac.yaml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+build_prefix: extra-x86_64
+maintainers:
+- github: kiri2002
+  email: kiri@vern.cc
+update_on:
+- regex: IBS_([.]*)_unix_[0-9]{8}.tar.gz
```
Comment (sukanka) [inline BioArchLinux/ibs/lilac.yaml:6]: this regex does not match the current pkgver, please recheck

**Comment by sukanka** [comment]
@starsareintherose , do you have any suggestion?
I have no more suggestions and think it's now ready to get merged

**Comment by Kiri2002** [comment]
> > Another thing to mention is that, we (maybe just I) prefer files to be placed under `/usr` instead of `/opt`, unless it's really a mess.
> > As it is a java package, I strongly recommend you to follow [Java package guidelines](https://wiki.archlinux.org/title/Java_package_guidelines)
> 
> Also am I. I would like to put it under /usr instead of /opt, I have packaged one pkg called pleroma which has quite a lot lib and so on, then I put it under /opt/pleroma, it's a messy path.

**Comment by starsareintherose** [comment]
It's okay, make sure your email is not a Microsoft mail then we can merge

File: BioArchLinux/ibs/PKGBUILD [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+        "$pkgname.desktop"
+        "$pkgname.sh")
+sha256sums=('2d8db720b5071f534cd49a1a29f6f5baa81966e436e5d19c492d08dfb42dcfbe'
+            'fe44d735f00d2b7485079ce995907640f3cfe18008d21664f21a830e6aadcff7'
+            'bda84c612959d73e0438b0d8eb582c1d9b3795ac311fe369ee3554fbb16a5895')
+package() {
+	mkdir -p $pkgdir/usr/share/$pkgname
+	mkdir -p $pkgdir/usr/share/doc/$pkgname
+	cp -pvr $srcdir/IBS/* $pkgdir/usr/share/$pkgname
+	mv $pkgdir/usr/share/$pkgname/'IBS 1.0.3' $pkgdir/usr/share/$pkgname/$pkgname
```
Comment (hubutui) [inline BioArchLinux/ibs/PKGBUILD:20]: `1.0.3` is pkgver? What happen if this pkg upgrade?

**Comment by hubutui** [comment]
BTW, it would be great if you squash all these commits into one before merging it to the master branch. I'm not sure if the `squash and merge` button from GitHub works as expected.

**Comment by Kiri2002** [comment]
> It's okay, make sure your email is not a Microsoft mail then we can merge

sure not, it's kiri@vern.cc, by the way, vern.cc is a free conmunity offer libre net service

**Comment by starsareintherose** [comment]
Please check your email which registered GitHub, invitation email is sent

**Pull request merged**